frontend/class-json-ld.php

```diff
@@ -0,0 +1,270 @@
+	 */
+	protected function output( $context ) {
+		/**
+		 * Filter: 'wpseo_json_ld_output' - Allows filtering of the JSON+LD output
+		 *
+		 * @api array $output The output array, before its JSON encoded
+		 *
+		 * @param string $context The context of the output, useful to determine whether to filter or not
+		 */
+		$this->data = apply_filters( 'wpseo_json_ld_output', $this->data, $context );
+
+		if ( function_exists( 'wp_json_encode' ) ) {
+			$json_data = wp_json_encode( $this->data );  // wp_json_encode was introduced in WP 4.1
+		}
+		else {
+			$json_data = json_encode( $this->data );
+		}
+
+		if ( is_array( $this->data ) && ! empty( $this->data ) ) {
+			echo "<script type='application/ld+json'>", $json_data, '</script>', "\n";
+		}
+
+		// Empty the $data array so we don't output it twice.
+		$this->data = array();
+	}
+
+	/**
+	 * Schema for Organization
+	 */
+	private function organization() {
+		if ( '' !== $this->options['company_name'] ) {
+			$this->data['@type'] = 'Organization';
+			$this->data['name']  = $this->options['company_name'];
+			$this->data['logo']  = $this->options['company_logo'];
+			return;
+		}
+		$this->data = false;
+	}
+
+	/**
+	 * Schema for Person
+	 */
+	private function person() {
+		if ( '' !== $this->options['person_name'] ) {
+			$this->data['@type'] = 'Person';
+			$this->data['name']  = $this->options['person_name'];
+			return;
+		}
+		$this->data = false;
+	}
+
+	/**
+	 * Prepares the organization or person markup
+	 */
+	private function prepare_organization_person_markup() {
+		$this->fetch_social_profiles();
+
+		$this->data = array(
+			'@context' => 'http://schema.org',
+			'@type'    => '',
+			'url'      => WPSEO_Frontend::get_instance()->canonical( false, true ),
+			'sameAs'   => $this->profiles,
+		);
+	}
+
+	/**
+	 * Outputs code to allow Google to recognize social profiles for use in the Knowledge graph
+	 *
+	 * @since 1.8
+	 */
+	public function organization_or_person() {
+		if ( ! is_front_page() || '' === $this->options['company_or_person'] ) {
+			return;
+		}
+
+		$this->prepare_organization_person_markup();
+
+		switch ( $this->options['company_or_person'] ) {
+			case 'company':
+				$this->organization();
+				break;
+			case 'person':
+				$this->person();
+				break;
+		}
+
+		$this->output( $this->options['company_or_person'] );
+	}
+
+	/**
+	 * Retrieve the social profiles to display in the organization output.
+	 *
+	 * @since 1.8
+	 *
+	 * @link  https://developers.google.com/webmasters/structured-data/customize/social-profiles
+	 */
+	private function fetch_social_profiles() {
+		$social_profiles = array(
+			'facebook_site',
+			'instagram_url',
+			'linkedin_url',
+			'plus-publisher',
+			'myspace_url',
+			'youtube_url',
+			'pinterest_url',
+		);
+		foreach ( $social_profiles as $profile ) {
+			if ( $this->options[ $profile ] !== '' ) {
+				$this->profiles[] = $this->options[ $profile ];
```

Comment:
You can return the profiles as well, instead of storing them in a property that is used only in one method.